datasafe-encryption/datasafe-encryption-impl/src/main/java/de/adorsys/datasafe/encrypiton/impl/cmsencryption/CMSEncryptionServiceImpl.java

```diff
@@ -57,18 +60,36 @@ public CMSEncryptionServiceImpl(ASNCmsEncryptionConfig encryptionConfig) {
     @Override
     @SneakyThrows
     public OutputStream buildEncryptionOutputStream(OutputStream dataContentStream,
-                                                    Set<PublicKeyIDWithPublicKey> publicKeys) {
+                                                    Set<PublicKeyIDWithPublicKey> publicKeys,
+                                                    KeyPair senderKeyPair) {
         return streamEncrypt(
                 dataContentStream,
-                publicKeys.stream().map(
-                        it -> new JceKeyTransRecipientInfoGenerator(
-                                it.getKeyID().getValue().getBytes(), it.getPublicKey()
-                        )
-                ).collect(Collectors.toSet()),
+                publicKeys.stream().map(it -> getRecipientInfoGenerator(it, senderKeyPair)).collect(Collectors.toSet()),
                 encryptionConfig.getAlgorithm()
         );
     }
 
+    private RecipientInfoGenerator getRecipientInfoGenerator(PublicKeyIDWithPublicKey keyWithId, KeyPair senderKeyPair) {
+        if ("RSA".equals(keyWithId.getPublicKey().getAlgorithm())) {
+            return new JceKeyTransRecipientInfoGenerator(keyWithId.getKeyID().getValue().getBytes(), keyWithId.getPublicKey());
+        }
+        if (Set.of("ECDH", "EC").contains(keyWithId.getPublicKey().getAlgorithm())) {
+            return getJceKeyAgreeRecipientInfoGenerator(senderKeyPair, keyWithId);
+        }
+        return null;
+    }
+
+    @SneakyThrows
+    public JceKeyAgreeRecipientInfoGenerator getJceKeyAgreeRecipientInfoGenerator(KeyPair senderKeyPair, PublicKeyIDWithPublicKey publicKeyWithId) {
+        var jceKeyAgreeRecipientInfoGenerator = new JceKeyAgreeRecipientInfoGenerator(
```

Comment:
why not use AES256_WRAP
```suggestion
        var jceKeyAgreeRecipientInfoGenerator = new JceKeyAgreeRecipientInfoGenerator(
                CMSAlgorithm.ECDH_SHA256KDF,
                senderKeyPair.getPrivate(),
                senderKeyPair.getPublic(),
                CMSAlgorithm.AES256_WRAP);

```

Reply:
Fixed with [99891c5](https://github.com/adorsys/datasafe/pull/234/commits/99891c578e4a10186e56d06a3646b4812e99ccf5)